numpy/lib/function_base.py

```diff
@@ -2699,6 +2717,14 @@ def __init__(self, pyfunc, otypes=None, doc=None, excluded=None,
         else:
             self._in_and_out_core_dims = None
 
+    def __get__(self, instance, owner):
+        return type(self)(self.pyfunc.__get__(instance, owner),
```

Comment:
Creating a vectorize instance can be moderately expensive (at least compared to calling it). Can we add caching on the calling object? I think something like `cached_property` would work: https://www.pydanny.com/cached-property.html

Reply:
Actually this mostly won't work because you don't necessarily know the name under which the attribute is accessed (well on recent Pythons you have `__set_name__`, but that's not something we can rely on).
I think keeping this working in all cases is more important that the optimization.

Reply:
You can work around that with a dict keyed by instance and unbound method. The bigger problem though is doing that without creating a reference cycle, which doesn't seem possible to me.

Reply:
Honestly, that seems to be a ridiculous overengineering.  Also, what if `self` is not hashable?

Reply:
Either way, I agree that caching is not a requirement here.